src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1219,6 +1227,9 @@ internal TextLoader(IHostEnvironment env, Options options = null, IMultiStreamSo
                 }
             }
 
+            if (options.DecimalMarker != '.' && options.DecimalMarker != ',')
+                throw _host.ExceptUserArg(nameof(Options.DecimalMarker), "Decimal marker cannot be the '{0}' character. It must be '.' or ','.", options.DecimalMarker);
+            _decimalMarker = options.DecimalMarker;
```

Comment:
I think we would also want to check here that if Decimal Marker is "," and Separator includes "," then Quoting must be allowed (throw in case it isn't)